pkg/registry/admissionregistration/validatingadmissionpolicybinding/authz.go

```diff
@@ -85,26 +85,44 @@ func (v *validatingAdmissionPolicyBindingStrategy) authorize(ctx context.Context
 		}
 	}
 
+	var attrs authorizer.AttributesRecord
 	paramRef := binding.Spec.ParamRef
 
-	// require that the user can read (verb "get") the referred resource.
-	attrs := authorizer.AttributesRecord{
-		User:            user,
-		Verb:            "get",
-		ResourceRequest: true,
-		Name:            paramRef.Name,
-		Namespace:       paramRef.Namespace,
-		APIGroup:        apiGroup,
-		APIVersion:      apiVersion,
-		Resource:        resource,
+	//!NOTE: This applies a more restrictive requirement when using
+	// cluster-scoped params. Requires LIST of the resource paramKind.
+	// Can relax if we have access to discovery/restmapper to decide if
+	// paramKind is SlusterScoped so we can disambiguate the two cases.
+	if len(paramRef.Namespace) > 0 {
+		attrs = authorizer.AttributesRecord{
+			User:            user,
+			Verb:            "get",
+			ResourceRequest: true,
+			Name:            paramRef.Name,
```

Comment:
What is name is unset? Do we have a test for it?

Reply:
I've refactored the logic here a bit and added cases for each possibility used by authz_test `validPolicyBindings`